packages/engine/tests/simulations/state_edit_bool_dot/src/behaviors/test.js

```diff
@@ -0,0 +1,8 @@
+/**
+ * @param {AgentState} state of the agent
+ * @param {AgentContext} context of the agent
+ */
+const behavior = (state, context) => {
+  state.b = state.a && false;
+  state.a = false;
```

Comment:
Is there a reason we're resetting `state.a` here?

It might seem like a somewhat silly question, but I think it might be indicative that the tests aren't as self-explanatory as we'd like, and we should think about a way to fix that. (Even if it's just adding a "description" field or something in `integration-test.json`).

The reason I ask about us setting `state.a` is I'm a little cautious about us using false, as it's a default value and this test could possibly still pass if a number of things go wrong. Wouldn't having `state.b` and `state.a` be different things be a better test?

Reply:
(Also if we make changes can we do the same for the `.get` and `.set` tests)

Reply:
The tests for the different types are more or less the same. cc @nshlapo

Reply:
@Alfred-Mountfield This is testing both reading and writing to state. Assigning state.b is technically the 'read' part, while assigning state.a is the 'write' part.

Reply:
I can change the assignments to `true`.

Reply:
Apologies I thought Tim had followed-up this conversation inline, I think we got distracted by other PRs. We've resolved this on Slack, but for persistence:

The problem I failed to convey was just that I think using `false` for both isn't a good selected value, as there are ways we could mess up state where this would still pass. As such Tim has suggested:
```js
state.b = state.a && true;
state.a = false;
```
 and testing for
 ```js
"a": false,
"b": true
```

What do you think?

Reply:
We might also want to add a check for `typeof state.a == "boolean"`